### PR TITLE
Add data monitoring to dscp-identity-service

### PR DESCRIPTION
**Pull request by nitro-marky**
Uses the DSCP-IPFS monitoring work done by @n3op2 to add the prom-bundle for metric collection. It also clears the prom client when the server is made to fix tests and ensures `Error: Cannot add the default metrics twice to the same registry` doesn't appear.

### CODE REVIEW

File: test/integration/httpServer.test.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ describe('health', function () {
     app = await createHttpServer()
   })
 
+  afterEach(function () {})
+
```
Comment (jonmattgray) [inline test/integration/httpServer.test.js:16]: remove?

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
```
------------------------------------------------------------------------------------------------------------------------
==> Logs of container dscp-identity-service-knon39rwod-95b54cd77-nnmdc
------------------------------------------------------------------------------------------------------------------------
node:internal/errors:478
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'express-prom-bundle' imported from /dscp-identity-service/app/server.js
    at new NodeError (node:internal/errors:387:5)
    at packageResolve (node:internal/modules/esm/resolve:852:9)
    at moduleResolve (node:internal/modules/esm/resolve:901:20)
    at defaultResolve (node:internal/modules/esm/resolve:1115:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:837:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

See why this is erroring.

**Comment by nitro-marky** [comment]
It seems to be passing now, I had already installed that package so don't know why it was failing.

File: package.json [hunk 2]
```diff
@@ -58,13 +58,15 @@
     "eslint": "^8.25.0",
     "eslint-config-prettier": "^8.5.0",
     "eslint-plugin-prettier": "^4.2.1",
+    "express-prom-bundle": "^6.5.0",
```
Comment (mattdean-digicatapult) [inline package.json:61]: should this not be a `dependency` rather than a `devDependency`?

File: package.json [hunk 3]
```diff
@@ -58,13 +58,15 @@
...
     "mocha": "^10.1.0",
     "mock-jwks": "^1.0.5",
     "nock": "^13.2.9",
     "nodemon": "^2.0.20",
     "nyc": "^15.1.0",
     "pino-colada": "^2.2.2",
     "prettier": "^2.7.1",
+    "prom-client": "^14.1.0",
```
Comment (mattdean-digicatapult) [inline package.json:69]: should this not be a `dependency` rather than a `devDependency`?

**Review by mattdean-digicatapult — DISMISSED** [review]
lgtm

File: app/server.js [hunk 4]
```diff
@@ -27,6 +29,18 @@ export async function createHttpServer() {
   app.use(compression())
   app.use(bodyParser.json())
 
+  client.register.clear()
+  app.use(
+    promBundle({
+      includePath: true,
+      promClient: {
+        collectDefaultMetrics: {
+          prefix: 'dscp_identity_service_',
```
Comment (n3op2) [inline app/server.js:38]: can we just say identity_service_ to be inline with ipfs

**Review by n3op2 — CHANGES_REQUESTED** [review]

File: app/server.js (hunk 4 above)
Comment (n3op2) [reply app/server.js:38]: https://github.com/digicatapult/dscp-ipfs/blob/main/app/server.js#L36

Comment (nitro-marky) [reply app/server.js:38]: No worries, I was told to put `dscp_identity_service_`, I'll change it

Comment (dblane-digicatapult) [reply app/server.js]: No leave it as `dscp-identity-service`, the ipfs one needs to be changed.

**Review by n3op2 — APPROVED** [review]

**Review by n3op2 — DISMISSED** [review]

File: app/server.js [hunk 5]
```diff
@@ -27,6 +29,18 @@ export async function createHttpServer() {
   app.use(compression())
   app.use(bodyParser.json())
 
+  client.register.clear()
+  app.use(
+    promBundle({
+      includePath: true,
+      promClient: {
+        collectDefaultMetrics: {
+          prefix: 'dscp_identity_service',
```
Comment (n3op2) [inline app/server.js:38]: you need underscore at the end otherwise it will look like `your_namesomemetricname`

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**